src/fping.c

```
@@ -970,6 +978,11 @@ int main(int argc, char **argv)
         exit(1);
     }
 
+    if (json_flag && !count_flag) {
+        fprintf(stderr, "%s: with -J or --json required -c or -C\n", prog);
+        exit(1);
+    }
+
```

How about adding another check for options that are either ignored with `-J` or produce non-JSON output despite `-J`, and exit if this is detected?

---

I think this is just `--timestamp-format` and `-e, --elapsed` (both are silently ignored). For `-e`, this could just be documented. For `--timestamp-format`, I'd say JSON output could in general support the different formats, and I'd prefer `fping` to reject combining the `--json` and `--timestamp-format` options for now.

---

I have rewritten the --timestamp-format a bit for the JSON output. But I'm not sure whether I haven't built in another security problem.

If not and this is so good, I would extend the tests even further

---

But the behavior with -e is identical to without JSON output and because the JSON output requires -c or -C, this fits from my point of view

---

You are right, `-e` is silently ignored for `-cN` and `-CN`, so it should still be silently ignored when adding `-J`.